library/Rules/Email.php

```diff
@@ -16,13 +16,30 @@
 use Egulias\EmailValidator\EmailValidator;
 use Egulias\EmailValidator\Validation\RFCValidation;
 
-class Email extends AbstractRule
+/**
+ * Validates an email address.
+ *
+ * @author Eduardo Gulias Davis <me@egulias.com>
+ * @author Henrique Moody <henriquemoody@gmail.com>
+ * @author Paul Karikari <paulkarikari1@gmail.com>
+ */
+final class Email extends AbstractRule
 {
+    /**
+     * Assigns email validator.
+     *
+     * @param EmailValidator $emailValidator
+     */
     public function __construct(EmailValidator $emailValidator = null)
     {
         $this->emailValidator = $emailValidator;
     }
 
+    /**
+     * Gets email validator.
+     *
+     * @return EmailValidator | null
+     */
     public function getEmailValidator()
     {
         if (class_exists(EmailValidator::class)
```

Comment:
Can you please declare the use of functions `is_string`, `filter_var` and `class_exists`